### PR TITLE
Add IaC scripts

### PR DESCRIPTION
**Pull request by PasanT9**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: sample_input.tfvars [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+eks_cluster_node_memory_utilization_warning_threshold      = 80
+eks_cluster_container_cpu_utilization_warning_threshold    = 80
+eks_cluster_container_memory_utilization_warning_threshold = 80
+eks_cluster_container_restart_warning_threshold            = 20
+alert_subscribers = [
+  {
+    "protocol" : "email"
+    "endpoint" : "user@wso2.com"
+  }
+]
```
Comment (ThilinaManamgoda) [inline sample_input.tfvars]: Let's add a new line

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      "from_port" : 443
+      "to_port" : 443
+      "protocol" : "-1"
+      "cidr_blocks" : ["${var.bastion_ip_address}/32"]
+      "security_groups" : []
+    }
+  ]
+  tags       = var.default_tags
+  depends_on = []
+}
```
Comment (ThilinaManamgoda) [inline main.tf]: Let's add a new line

File: alerts.tf [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+#
+# This software is the property of WSO2 LLC. and its suppliers, if any.
+# Dissemination of any information or reproduction of any material contained
+# herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# --------------------------------------------------------------------------------------
+
+module "alert_group" {
+  source      = "git::https://github.com/wso2/aws-terraform-modules.git//modules/aws/SNS-Topic"
```
Comment (ThilinaManamgoda) [inline alerts.tf]: Shall we create a tag from this repo and refer to a particular tag. This is applicable for all the `source` references.

File: alerts.tf [hunk 4]
```diff
@@ -0,0 +1,127 @@
+# -------------------------------------------------------------------------------------
+#
+# Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com). All Rights Reserved.
```
Comment (ThilinaManamgoda) [inline alerts.tf]: Let's change the license header year to 2024

File: locals.tf [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  container_cpu_utilization_metric_name_usage    = "container-cpu-usage"
+  container_memory_utilization_metric_name_usage = "container-memory-usage"
+  container_restarts_metric_name_usage           = "container_number_of_restarts"
+
+
+  cloud_watch_average_statistic_string = "Average"
+  cloud_watch_sum_statistic_string     = "Sum"
+
+  cloud_watch_gte_metrics_threshold_string = "GreaterThanOrEqualToThreshold"
+}
```
Comment (ThilinaManamgoda) [inline locals.tf]: Let's add a new line

**Review by ThilinaManamgoda — CHANGES_REQUESTED** [review]

File: sample_input.tfvars
Comment (athiththan11) [inline sample_input.tfvars:1]: Shall we keep the sample files in a separate directory and point these sample files in the READ ME to get a copy, rename and fill with required configurations? 

Suggestion: Create a directory named `example` in the root directory of this project and keep the sample configuration file, and  make a note in the READ ME to take a copy and make necessary changes

**Review by athiththan11 — CHANGES_REQUESTED** [review]
https://github.com/wso2/iac-aws-wso2-products/pull/1/files#r1441388788

File: README.md [hunk 6]
```diff
@@ -1,2 +1,129 @@
...
+| Name | Description |
+|------|-------------|
+| <a name="output_bastion_instance"></a> [bastion\_instance](#output\_bastion\_instance) | ID of the bastion instance. |
+| <a name="output_database_writer_endpoint"></a> [database\_writer\_endpoint](#output\_database\_writer\_endpoint) | Writer endpoint of the database instance. |
+| <a name="output_efs_efs_access_point"></a> [efs\_efs\_access\_point](#output\_efs\_efs\_access\_point) | ID of the EFS Access Point |
+| <a name="output_efs_id"></a> [efs\_id](#output\_efs\_id) | ID of the Elastic File Storage |
+| <a name="output_filestore_location"></a> [filestore\_location](#output\_filestore\_location) | Location of the filestore. |
+
+---
+This document is autogenerated using terraform-docs (https://github.com/terraform-docs/terraform-docs)
```
Comment (athiththan11) [inline README.md]: We can remove this portion and add instructions on how to use the modules and what are the prerequisites to be done.

File: sample_input.tfvars (hunk 1 above)
Comment (PasanT9) [reply sample_input.tfvars]: Ack. Fixed

File: main.tf (hunk 2 above)
Comment (PasanT9) [reply main.tf]: Ack. Fixed

File: alerts.tf (hunk 3 above)
Comment (PasanT9) [reply alerts.tf]: Ack. Now we are using the tag v1.12.0 which contains latest changes.

File: alerts.tf (hunk 4 above)
Comment (PasanT9) [reply alerts.tf]: Fixed here and other places as well.

File: locals.tf (hunk 5 above)
Comment (PasanT9) [reply locals.tf]: Acl. Fixed.

File: sample_input.tfvars
Comment (PasanT9) [reply sample_input.tfvars:1]: Ack. Fixed

File: README.md (hunk 6 above)
Comment (PasanT9) [reply README.md]: The ReadMe are now upgraded with instructions.

**Review by ThilinaManamgoda — DISMISSED** [review]

File: database.tf [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  }
+  database_name           = var.db_primary_db_name
+  master_username         = var.db_master_username
+  master_password         = var.db_password
+  engine                  = var.db_engine
+  engine_version          = var.db_engine_version
+  engine_mode             = var.db_engine_mode
+  db_subnet_group_name    = module.db_subnet_group.subnet_group_name
+  backup_retention_period = 7 # TODO parameterize
```
Comment (athiththan11) [inline database.tf]: Why do we need this TODO comments?

File: locals.tf [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  warning_alert_usage_suffix  = "warning"
+  critical_alert_usage_suffix = "critical"
+
+  node_cpu_utilization_metric_name_usage         = "node-cpu-usage"
+  node_memory_utilization_metric_name_usage      = "node-memory-usage"
+  container_cpu_utilization_metric_name_usage    = "container-cpu-usage"
+  container_memory_utilization_metric_name_usage = "container-memory-usage"
+  container_restarts_metric_name_usage           = "container_number_of_restarts"
+
+
```
Comment (athiththan11) [inline locals.tf]: Let's remove this extra line

File: sample/secret.auto.tfvars
Comment (athiththan11) [inline sample/secret.auto.tfvars:1]: Let's add the License Header to this as well

**Review by athiththan11 — CHANGES_REQUESTED** [review]

**Review by athiththan11 — APPROVED** [review]

**Pull request merged**